feat: add a cancel function return from WatchConfig() to stop watchin…

chowyi opened the pull request:
I need to stop the WatchConfig() function in certain scenarios. I found someone with the same need in an open Issue: https://github.com/spf13/viper/issues/1404. I have attempted to implement it.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/spf13/viper?pullRequest=1583) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/spf13/viper?pullRequest=1583) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/spf13/viper?pullRequest=1583) it.</sub>

sagikazarmark commented:
Thanks @chowyi!

Unfortunately, this is a breaking change which is not allowed on the v1 release track at the moment.

I'd be happy to accept a change that introduces a new function implementing the same functionality.

chowyi commented:
Thanks for your suggestion. 
I reverted the previous modification, added a new function `StopWatching()` implementing the same functionality and the unitTest.
